core/src/main/scala/com/pingcap/tispark/TiBatchWrite.scala

```diff
@@ -146,7 +151,26 @@ class TiBatchWrite(@transient val df: DataFrame,
 
     // lock table
     tiDBJDBCClient = new TiDBJDBCClient(TiDBUtils.createConnectionFactory(options.url)())
-    isEnableTableLock = tiDBJDBCClient.isEnableTableLock
+    isEnableTableLock = {
+      if (tiDBJDBCClient.isEnableTableLock) {
+        if (tiDBJDBCClient.getDelayCleanTableLock >= MIN_DELAY_CLEAN_TABLE_LOCK) {
```

Comment:
Since we have logic to check “delay clean table lock” here, I think it’s better to make MIN_DELAY_CLEAN_TABLE_LOCK configurable.

Reply:
pls give a reason, why make it configurable?

Reply:
Because delayCleanTableLock is configurable in TiDB, the user may want to reduce table lock duration by TiDB configuration.

Reply:
not necessary as discussed